R/fread.R

```diff
@@ -26,44 +26,60 @@ fread <- function(input="",file,sep="auto",sep2="auto",dec=".",quote="\"",nrows=
   stopifnot(nThread>=1L)
   if (!missing(file)) {
     if (!identical(input, "")) stop("You can provide 'input=' or 'file=', not both.")
-    if (!file.exists(file)) stop("File '",file,"' does not exist.")
-    if (isTRUE(file.info(file)$isdir)) stop("File '",file,"' is a directory. Not yet implemented.") # dir.exists() requires R v3.2+, #989
+    file_info = file.info(file)
+    if (is.na(file_info$size)) stop("File '",file,"' does not exist or is non-readable.")
```

Comment:
`is.na(file.info(file)$size)` should be equivalent to `!file.exists(file)`, as per my reading of `?file.info`:

> Entries for non-existent or non-readable files will be NA.

Switching to this since there are now three checks, all of which can be run through the single call to `file.info`: (1) file exists? (2) file is directory? (3) file non-empty?